src/runtime-rs/crates/resource/src/network/endpoint/vhost_user_endpoint.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright (C) 2019-2023 Ant Group. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::path::Path;
+use std::sync::Arc;
+
+use anyhow::{anyhow, Context, Result};
+use async_trait::async_trait;
+use hypervisor::device::device_manager::{do_handle_device, DeviceManager};
+use hypervisor::device::{DeviceConfig, DeviceType};
+use hypervisor::{Hypervisor, VhostUserConfig, VhostUserNetDevice, VhostUserType};
+use tokio::sync::RwLock;
+
+use super::endpoint_persist::VhostUserEndpointState;
+use super::Endpoint;
+use crate::network::EndpointState;
+
+/// VhostUserEndpoint uses vhost-user-net device, which supports DPDK, etc.
+#[derive(Debug)]
+pub struct VhostUserEndpoint {
+    // Endpoint index
+    #[allow(dead_code)]
```

Comment:
why dead code ?

Reply:
I removed this field from `VhostUserEndpoint`.